sdk/core/azure-json/src/main/java/com/azure/json/JsonNull.java

```diff
@@ -0,0 +1,44 @@
+package com.azure.json;
+
+import java.io.IOException;
+
+/**
+ * Class representing the JSON null type
+ */
+public class JsonNull extends JsonElement{
```

Comment:
Since there can only be one kind of null I would suggest making this class `final` and making the only constructor is has private and just provide a constant.

```java
public final class JsonNull extends JsonElement {
    public static final INSTANCE = new JsonNull();

    private JsonNull() { }

    // other implementation details
}
```

This helps in two ways, the equality operator (`==`) between two `JsonNull`s will always be true and the is no need to instantiate `JsonNull`s which would always be the same.

And looping back to `JsonBoolean` the same could be done for that but having two constants `TRUE` and `FALSE`, getting the same benefits I just mentioned.